src/Chapter2.hs

```diff
@@ -841,7 +879,13 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [Int] -> [Int]
+rotate _ [] = []
+rotate _ [x] = [x]
+rotate times l
+  | times < 0 = []
+  | times == 0 = l
+  | otherwise = subList times (times + length l - 1) (cycle l)
```

Comment:
Well done on reusing the `subList` function in `rotate`! So far, you are the first one :1st_place_medal: